openapi.yaml

```diff
@@ -928,12 +936,29 @@ components:
         referencePath:
           type: string
           description: Where to find the reference in the Bundle entry
-          example: 'provider.reference'
+          example: provider.reference
       description: A connection between entries in a Bundle
-    ResourceRequest:
+    ResourceCreateRequest:
+      required:
+        - resourceType
+      type: object
+      additionalProperties:
+        type: string
       properties:
-        resource:
-          type: object
+        resourceType:
+          type: string
+    ResourceUpdateRequest:
+      required:
+        - id
+        - resourceType
+      type: object
+      additionalProperties:
+        type: string
+      properties:
+        id:
+          type: string
+        resourceType:
+          type: string

```

Comment:
Doesn't look right. I'm no OpenAPI/Swagger expert but I think this means that all fields of the object must be strings. FHIR resources do not conform to this schema(they can have nested objects, arrays, etc.)

Have you tested this schema changes?

Reply:
I had tested the schema change and couldn't find a way to produce a dictionary with dynamic keys and values that were `string`, `array`, and `objects`. 
Here's some of the examples I tried
```
ResourceCreateRequest:
    required:
        - resourceType
    type: object
    additionalProperties: true
    properties:
        resourceType:
            type: string

Produces

{
  "resourceType": "string",
  "additionalProp1": {}
}

----

ResourceCreateRequest:
    required:
        - resourceType
    type: object
    additionalProperties:
        type: object
        properties:
            code:
                type: integer
            text:
                type: string
    properties:
        resourceType:
            type: string
    
produces

{
  "resourceType": "string",
  "additionalProp1": {
    "code": 0,
    "text": "string"
  },
  "additionalProp2": {
    "code": 0,
    "text": "string"
  },
  "additionalProp3": {
    "code": 0,
    "text": "string"
  }
}


ResourceCreateRequest:
    required:
        - resourceType
    type: object
    additionalProperties:
        type: string
    properties:
        resourceType:
            type: string

produces

{
  "resourceType": "string",
  "additionalProp1": "string",
  "additionalProp2": "string",
  "additionalProp3": "string"
}


ResourceCreateRequest:
      required:
        - resourceType
      type: object
      properties:
        resourceType:
          type: string
        additionalProp1:
          type: string
        additionalProp2:
          type: object
        additionalProp3:
          type: array
          items:
            oneOf:
              - type: string
              - type: integer
              - type: object

produces


{
  "resourceType": "string",
  "additionalProp1": "string",
  "additionalProp2": {},
  "additionalProp3": [
    "string",
    0,
    {}
  ]
}
```

The last example produces 
```
"additionalProp1": "string",
  "additionalProp2": {},
  "additionalProp3": [
    "string",
    0,
    {}
  ]
```
This is what I'm hoping Swagger could generate, but I had to do it manually to get that format. I think that's hacky and probably isn't to spec.

Base on the documentation `If the dictionary values can be of any type (aka free-form object), use additionalProperties: true:`. So since the values of a resource can be of any type, this might be our best bet. I'll change the code to reflect that.

Reply:
`additionalProperties: true` looks like the way to go